dmcontent/markdown.py

```diff
@@ -0,0 +1,42 @@
+from typing import Mapping
+from xml.etree.ElementTree import Element
+
+from markdown.extensions import Extension
+from markdown.treeprocessors import Treeprocessor
+
+
+class _ClassAddingTreeprocessor(Treeprocessor):
+    _mapping: Mapping[str, str]
+
+    def __init__(self, mapping: Mapping[str, str]):
+        self._mapping = mapping
+
+    def run(self, root: Element):
+        self._run(root)
+
+    def _run(self, element: Element):
+        # recurse
+        for child in element:
+            self._run(child)
+
+        # mutate each affected element in-place
+        if element.tag in self._mapping:
+            orig_class = element.attrib.get("class", "")
+            element.set("class", (orig_class + " " if orig_class else "") + self._mapping[element.tag])
+
+
+class GovUkFrontendExtension(Extension):
+    """
+    Markdown extension adding govuk-frontend classes to generated paragraphs and lists (note - will not affect
+    manual html)
```

Comment:
What happens here with escaping? Could we add a test?

Reply:
Oh I thought I included an ampersand in the test text but I didn't. I'll do that.

Reply:
Done